src/helpers/simplify_claims.ts

```diff
@@ -1,176 +1,166 @@
+import type { CustomSimplifiedSnak, SimplifiedClaim, SimplifiedClaims, SimplifiedPropertyClaims, SimplifiedPropertySnaks, SimplifiedSnaks, SimplifyClaimsOptions, SimplifySnakOptions, SimplifySnaksOptions } from '../types/simplify_claims.js'
+
+/**
+ * Tries to replace wikidata deep snak object by a simple value
+ * e.g. a string, an entity Qid or an epoch time number
+ * Expects a single snak object
+ * Ex: entity.claims.P369[0]
+ */
+export function simplifySnak (snak: Snak, options: SimplifySnakOptions = {}) {
+  const { keepTypes, keepSnaktypes, keepHashes } = parseKeepOptions(options)
+
+  let value
+  const { datatype, datavalue, snaktype, hash } = snak
 
   if (datavalue) {
-    value = parseClaim(datatype, datavalue, options, claim.id)
+    value = parseSnak(datatype, datavalue, options)
   } else {
     if (snaktype === 'somevalue') value = options.somevalueValue
     else if (snaktype === 'novalue') value = options.novalueValue
     else throw new Error('no datavalue or special snaktype found')
   }
 
-  // Qualifiers should not attempt to keep sub-qualifiers or references
-  if (isQualifierSnak) {
-    if (!(keepHashes || keepTypes || keepSnaktypes)) return value
-
-    const valueObj: any = { value }
-
-    if (keepHashes) valueObj.hash = claim.hash
+  // No need to test keepHashes as it has no effect if neither
+  // keepQualifiers or keepReferences is true
+  if (keepTypes || keepSnaktypes || keepHashes) {
+    // When keeping qualifiers or references, the value becomes an object
+    // instead of a direct value
+    const valueObj: CustomSimplifiedSnak = { value }
     if (keepTypes) valueObj.type = datatype
     if (keepSnaktypes) valueObj.snaktype = snaktype
-
+    if (keepHashes) valueObj.hash = hash
     return valueObj
+  } else {
+    return value
   }
-  if (isReferenceSnak) {
-    if (!keepTypes) return value
+}
+
+export function simplifyClaim (claim: Claim, options: SimplifySnakOptions = {}): SimplifiedClaim {
+  const { keepQualifiers, keepReferences, keepIds, keepTypes, keepSnaktypes, keepRanks } = parseKeepOptions(options)
+
+  const { mainsnak, rank } = claim
+
+  const value = simplifySnak(mainsnak, options)
 
-    return { type: datatype, value }
-  }
   // No need to test keepHashes as it has no effect if neither
   // keepQualifiers or keepReferences is true
   if (!(keepQualifiers || keepReferences || keepIds || keepTypes || keepSnaktypes || keepRanks)) {
     return value
   }
 
-  // When keeping qualifiers or references, the value becomes an object
-  // instead of a direct value
-  const valueObj: any = { value }
-
-  if (keepTypes) valueObj.type = datatype
-
-  if (keepSnaktypes) valueObj.snaktype = snaktype
+  // When keeping other attributes, the value becomes an object instead of a direct value
+  let valueObj
+  if (isPlainObject(value)) {
+    valueObj = value as CustomSimplifiedClaim
+  } else {
+    valueObj = { value } as CustomSimplifiedClaim
+  }
```

Comment:
It should be like the following suggestion but that's currently not possible as it results in a bunch of type errors…

```diff
-  let valueObj
-  if (isPlainObject(value)) {
-    valueObj = value as CustomSimplifiedClaim
-  } else {
-    valueObj = { value } as CustomSimplifiedClaim
-  }
+  let valueObj: CustomSimplifiedClaim
+  if (isPlainObject(value)) {
+    valueObj = value
+  } else {
+    valueObj = { value }
+  }
```

For now I would suggest to just type it explicitly as any so its noticed easier later on and can be fixed on its own rather than in this PR.

```suggestion
  let valueObj: any // TODO: type as CustomSimplifiedClaim
  if (isPlainObject(value)) {
    valueObj = value
  } else {
    valueObj = { value }
  }
```